app/services/schools/advice/base_service.rb

```diff
@@ -0,0 +1,26 @@
+module Schools
+  module Advice
+    class BaseService
+      def initialize(school, meter_collection)
+        @school = school
+        @meter_collection = meter_collection
+      end
+
+      def date_range(meter: aggregate_meter)
+        [meter.amr_data.start_date, meter.amr_data.end_date]
```

Comment:
Just a thought but maybe this could return an struct or openstruct here too? (so we can use something like`date_range.end_date` instead `date_range[1]` or assigning to separate `start_date` and `end_date` variables)

Reply:
I've just removed the method for now. It wasn't doing much work and its not really adding any clarity to the code.

So I've changed the controller to access the date range it needs directly, and the asof_date method to do the same.